compiler/rustc_span/src/span_encoding.rs

```diff
@@ -136,13 +136,13 @@ impl Span {
         }
 
         // Partially-interned or fully-interned format.
-        let index =
-            with_span_interner(|interner| interner.intern(&SpanData { lo, hi, ctxt, parent }));
-        let ctxt_or_parent_or_marker = if ctxt2 <= MAX_CTXT {
-            ctxt2 as u16 // partially-interned
+        let (ctxt_or_parent_or_marker, ctxt) = if ctxt2 <= MAX_CTXT {
+            (ctxt2 as u16, SyntaxContext::from_u32(u32::MAX)) // partially-interned
```

Comment:
What's the significance of `SyntaxContext::from_u32(u32::MAX)`? Is it a sentinel value?   Would it make sense to add some associated constant `SyntaxContext::MAX` or `SyntaxContext::SENTINEL` and assert that its value is not used (where necessary)?

Reply:
With https://github.com/rust-lang/rust/pull/125017/files#r1600578394 it can be any value, because it will never be read.
We could use zero here as well, but `u32::MAX` will look more obviously wrong if it leaks to somewhere else.

In theory, `u32::MAX` can be a valid ctxt value on 64 bit systems.